marshaler.go

```diff
@@ -163,23 +176,27 @@ func (enc *Encoder) encode(b []byte, ctx encoderCtx, v reflect.Value) ([]byte, e
 		b = strconv.AppendUint(b, v.Uint(), 10)
 	case reflect.Int64, reflect.Int32, reflect.Int16, reflect.Int8, reflect.Int:
 		b = strconv.AppendInt(b, v.Int(), 10)
-	default:
-		err = fmt.Errorf("unsupported encode value kind: %s", v.Kind())
-	}
-	if err != nil {
-		return nil, err
+	case reflect.Array, reflect.Chan, reflect.Complex128, reflect.Complex64,
+		reflect.Func, reflect.Invalid, reflect.Uintptr, reflect.UnsafePointer:
+		return nil, fmt.Errorf("encode(type %s): %w", v.Kind(), errUnsupportedValue)
 	}
 
 	return b, nil
 }
 
 func isNil(v reflect.Value) bool {
 	switch v.Kind() {
-	case reflect.Ptr, reflect.Interface, reflect.Map:
+	case reflect.Ptr, reflect.UnsafePointer, reflect.Interface, reflect.Map, reflect.Slice:
 		return v.IsNil()
-	default:
+	case reflect.Array, reflect.Bool, reflect.Chan, reflect.Complex128,
+		reflect.Complex64, reflect.Float32, reflect.Float64, reflect.Func,
+		reflect.Int, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Int8,
+		reflect.Invalid, reflect.String, reflect.Struct, reflect.Uintptr, reflect.Uint, reflect.Uint16,
+		reflect.Uint32, reflect.Uint64, reflect.Uint8:
```

Comment:
I was first just following the linter first, then I realized you might have missed some Kinds which actually implement IsNil.
So I thought it was a good idea to be exhaustive.
Now, after your comment, I went and actually benched the diff between having a default and not and...

There is a difference: about 0.5ns per op. It's extremely small, but it's also almost 50% slower, so it might not be so great in the end.

I'm shared between:
* keeping the linter as it it and adding a nolint (which will emphasize on the missing kinds at lint time)
* adding the conf you mentioned

Reply:
My personal opinion: A nolint comment feels like a false positive to me.  Linters that create a lot of "false positives" annoy me.  So, I'm inclined to disable linters that are more strict than I care do deal with.  The conf setting is a compromise that would keep me from disabling the linter altogether.  :smile:

Reply:
I think the initial use of `default` in those functions is warranted: here is a list of cases the lib support, for everything else (present or future): error. 

The whole reason the work of using the linters is to 1. catch dumb mistakes 2. provide initial feedback to contributors before sending the code review. As a result, I think we should err on the side of false negatives: maybe the linter will let an actual mistake through, but it is not going to be annoying and ask people to modify their code while a human reviewer would not have cared. In order to achieve that balance, we need to spend time trying out the different linters and options. In my opinion this is what is happening now, so I am very grateful for your time and patience @vincentserpoul  :)

In that light, I'd rather see the default case reverted and the option @moorereason mentioned turned on.